File: apps/fluster/lib/core/util/toggle_dark_mode.dart
import 'package:async_redux/async_redux.dart';
import 'package:fluster/core/state/nested_state/ui_state/actions/ui_actions.dart';
import 'package:fluster/core/state/store.dart';
import 'package:fluster/core/static/global_keys.dart';
import 'package:flutter/material.dart';

void toggleDarkMode(ThemeMode? providedThemeMode, BuildContext? context) {
    
  final themeMode =
      providedThemeMode ??
      Theme.of(context ?? desktopScaffoldMessenegerKey.currentState!.context)).brightness;
  if (context != null) {
    context.dispatch(
      SetThemeModeAction(
        (themeMode == Brightness.dark) ? ThemeMode.light : ThemeMode.dark,
      ),
    );
  } else {
    globalReduxStore.dispatch(
      SetThemeModeAction(
        (themeMode == Brightness.dark) ? ThemeMode.light : ThemeMode.dark,
      ),
    );
  }
}

File: apps/fluster/src/features/scaffold/state/actions/toggle_dark_mode.ts
import store from "@/state/store";
import { setThemeModeAction } from "../slice";
import { ThemeMode } from "../initial_state";
import { AppState } from "@/state/initial_state";
import { prefersDarkMode } from "#/scaffold/utils";

export const toggleDarkMode = () => {
  const currentThemeMode = (store.getState() as AppState).scaffold.themeMode;
  const themeMode =
    currentThemeMode === ThemeMode.system
      ? prefersDarkMode()
        ? ThemeMode.light
        : ThemeMode.dark
      : currentThemeMode === ThemeMode.light
      ? ThemeMode.dark
      : ThemeMode.light;
  document.body.classList[themeMode === ThemeMode.dark ? "add" : "remove"](
    "dark"
  );
  store.dispatch(setThemeModeAction(themeMode));
};
